### ISSUE TITLE
Catch error with circular categories nesting

### ISSUE THREAD

**Issue by briceparent** [Fantomas42/django-blog-zinnia#506]
# Actual behavior
I tried it in the demo site. If it's not up-to-date, maybe it has already been corrected.
If you edit category 1 (https://demo.django-blog-zinnia.com/admin/zinnia/category/1/change/) and define its mother category to Templates (for example), you get an error 500 (as Templates is a child of category 1).

# Expected behavior
You should get a ValidationError or ValueError (like the one you have when setting the mother category to the category itself), with an error messagedisplayed in the admin, instead of getting out of the admin with the error 500. 

# Steps to reproduce the issue
Go to https://demo.django-blog-zinnia.com/admin/zinnia/category/1/change/
  1. Set parent/mother category to Template
  2. Save
  3. -> error 500 (and lose the other changes you made during in the form)

# Specifications

 - Zinnia version: The one on the demo server
 - Django version: The one on the demo server
 - Python version: The one on the demo server
 - Operating system: The one on the demo server

# Disclaimer

Please read these questions **carefully** and answer honestly with an `x`
into all the boxes `[ ]`:

 - [x] I have read the guidelines for contributing linked to above.
 - [x] I have checked for duplicate issues.
 - [x] This is not a support request.

# More info
I'm not (yet) a user, I've just played with the demo site and the admin, and looked a bit in the code. And I'll probably use it as it seems a good solution to what I need.
If I do chose it, I'll probably write a test case for this issue, and make the correction if it hasn't been done already, but I might not be able to do that before some weeks.

- Brice

**Comment by Fantomas42** [comment]
Hello,

I have seen the traceback of the demo this morning, and I was surprised.
I was thinking that this case was handled.

I will take a look asap.

Regards

**Comment by Fantomas42** [comment]
Hello,

I have tried to reproduce both on local and demo site, I cannot reproduce the issue.

The error is correctly catched.

Regards

**Comment by briceparent** [comment]
Well, it's strange. I can't say with local server as I haven't got the time to check it out, but I've just tried again (using the link I posted above), and as I'm sure you've noticed in your notifications, it occurred again.
Maybe you're not using the demo user (demo:demo if I remember correctly) ?
And on local, maybe DEBUG = False changes the behaviour ?
Anyway, if I use this (I think I will) and it happens to me, I'll fix it and send a PR.
Thanks for your reactivity, and for this whole project
